packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,168 @@
+%lex
+%options case-insensitive
+
+%x string
+%x md_string
+%x title
+%x acc_title
+%x acc_descr
+%x acc_descr_multiline
+%s axis_data
+%s axis_band_data
+%s data
+%s data_inner
+%%
+\%\%(?!\{)[^\n]*                          /* skip comments */
+[^\}]\%\%[^\n]*                           /* skip comments */
+<axis_data>(\r?\n)                        { this.popState(); return 'NEWLINE'; }
+<data>(\r?\n)                             { this.popState(); return 'NEWLINE'; }
+[\n\r]+                                   return 'NEWLINE';
+\%\%[^\n]*                                /* do nothing */
+
+"title"                                   { return 'title'; }
+
+"accTitle"\s*":"\s*                         { this.pushState("acc_title");return 'acc_title'; }
+<acc_title>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_title_value"; }
+"accDescr"\s*":"\s*                         { this.pushState("acc_descr");return 'acc_descr'; }
+<acc_descr>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_descr_value"; }
+"accDescr"\s*"{"\s*                         { this.pushState("acc_descr_multiline");}
+<acc_descr_multiline>"{"                 { this.popState(); }
+<acc_descr_multiline>[^\}]*               { return "acc_descr_multiline_value"; }
+
+"xychart-beta"                            {return 'XYCHART';}
+(?:"vertical"|"horizontal")               {return 'CHART_ORIENTATION'}
+
+"x-axis"                                  { this.pushState("axis_data"); return "X_AXIS"; }
+"y-axis"                                  { this.pushState("axis_data"); return "Y_AXIS"; }
+<axis_data>"["                            { this.pushState("axis_band_data"); return 'SQUARE_BRACES_START'; }
+<axis_data>"-->"                          { return 'ARROW_DELIMITER'; }
+
+
+"line"                                    { this.pushState("data"); return 'LINE'; }
+"bar"                                     { this.pushState("data"); return 'BAR'; }
+<data>"["                                 { this.pushState("data_inner"); return 'SQUARE_BRACES_START'; }
+<axis_data,data_inner>[+-]?(?:\d+(?:\.\d+)?|\.\d+)   { return 'NUMBER_WITH_DECIMAL'; }
+<data_inner,axis_band_data>"]"            { this.popState(); return 'SQUARE_BRACES_END'; }
```

Comment:
It seems we push more states than pop, e.g. we push state with `line` and `bar` statements, but I don't see any places where we pop it. Somehow we must detect `line` or `bar` end. May be it is newline (then we must pop state at "\n") or closing bracket (if braces are necessary, then we could pop the state at "]").

Reply:
We do it at the top
https://github.com/subhash-halder/mermaid/blob/fae648c2538c62dba8d832309a9873916feb1ea5/packages/mermaid/src/diagrams/xychart/parser/xychart.jison#L18